src/core/document.js

```diff
@@ -261,6 +262,60 @@ class Page {
     );
   }
 
+  async saveNewAnnotations(handler, task, annotations) {
+    const partialEvaluator = new PartialEvaluator({
+      xref: this.xref,
+      handler,
+      pageIndex: this.pageIndex,
+      idFactory: this._localIdFactory,
+      fontCache: this.fontCache,
+      builtInCMapCache: this.builtInCMapCache,
+      standardFontDataCache: this.standardFontDataCache,
+      globalImageCache: this.globalImageCache,
+      options: this.evaluatorOptions,
+    });
+
+    if (this.xfaFactory) {
+      throw new Error("XFA: Cannot save any annotations");
+    }
```

Comment:
Let's move this to the *top* of the method instead, since none of the code above matters if we're going to throw here.
Also, how about the following error message instead (since I believe that we do support saving of e.g. existing forms in XFA document):
```suggestion
    if (this.xfaFactory) {
      throw new Error("XFA: Cannot save new annotations.");
    }
```